qt/scientific_interfaces/Indirect/ConvFit.cpp

```diff
@@ -1595,126 +1312,60 @@ void ConvFit::fitFunctionSelected(int fitTypeIndex) {
   }
 
-  auto fitFunctions = indexToFitFunctions(fitTypeIndex);
-  if (fitFunctions.isEmpty()) {
-    return;
-  }
-
-  auto lastFunction = fitFunctions.last();
-
-  // Remove previous parameters from tree
-  m_cfTree->removeProperty(m_properties["FitFunction1"]);
-  m_cfTree->removeProperty(m_properties["FitFunction2"]);
-
-  m_uiForm.ckPlotGuess->setChecked(false);
-
-  updatePlotOptions();
-
-  // Two Lorentzians Fit
-  if (lastFunction == "Lorentzian 2") {
-    m_properties["FitFunction1"] = m_grpManager->addProperty("Lorentzian 1");
-    m_cfTree->addProperty(m_properties["FitFunction1"]);
-    m_properties["FitFunction2"] = m_grpManager->addProperty("Lorentzian 2");
-    m_cfTree->addProperty(m_properties["FitFunction2"]);
-  } else {
-    m_properties["FitFunction1"] = m_grpManager->addProperty(lastFunction);
-    m_cfTree->addProperty(m_properties["FitFunction1"]);
-  }
-
-  addDefaultParametersToTree(fitFunctions);
-  updateProperties(selectedSpectrum());
-}
-
-/**
- * Adds all the parameters that are required for the fit function to the
- * parameter tree
- * @param fitFunction	:: The name of the fit function
- */
-void ConvFit::addDefaultParametersToTree(const QVector<QString> &fitFunctions) {
-
-  for (auto &fitFunction : fitFunctions) {
-    addDefaultParametersToTree(fitFunction);
-  }
+  setPropertyFunctions(indexToFitFunctions(fitTypeIndex));
+  fillPlotTypeComboBox(m_uiForm.cbPlotType);
 }
 
-void ConvFit::addDefaultParametersToTree(const QString &fitFunction) {
-  auto paramNames = getFunctionParameters(fitFunction);
-  bool isDelta = fitFunction == "Delta Function";
-  bool isTwoLorentzian = !isDelta && fitFunction == "Lorentzian 2";
-
-  for (auto &paramName : paramNames) {
-    QString propertyName = fitFunction + "." + paramName;
-    m_properties[propertyName] = m_dblManager->addProperty(paramName);
-
-    if (isTwoLorentzian) {
-      paramName = "default_" + paramName;
-    }
+QString ConvFit::addPrefixToParameter(const QString &parameter,
+                                      const QString &functionName,
+                                      const int &functionNumber) const {
+  QString tempPrefix = m_usedTemperature ? "f1." : "f0.";
 
-    if (m_defaultParams.contains(paramName)) {
-      m_dblManager->setValue(m_properties[propertyName],
-                             m_defaultParams[paramName]);
-    } else {
-      m_dblManager->setValue(m_properties[propertyName], 0);
-    }
-    m_dblManager->setDecimals(m_properties[propertyName], NUM_DECIMALS);
-
-    if (isTwoLorentzian) {
-      m_properties["FitFunction2"]->addSubProperty(m_properties[propertyName]);
-    } else if (!isDelta) {
-      m_properties["FitFunction1"]->addSubProperty(m_properties[propertyName]);
-    }
-  }
+  if (functionName == "Delta Function")
+    return "f1.f1.f" + QString::number(functionNumber) + "." + parameter;
+  else
+    return "f1.f1.f" + QString::number(functionNumber) + "." + tempPrefix +
+           parameter;
```

Comment:
is this always correct? Is it possible for a user to have a composite or product function?

Reply:
This is currently always correct for ConvFit.

This is because the following is always true:
'f0' is the location of the background within the composite function - always currently true, since it is always present (desirable to change this).
'f1' is the location of the convoluted function within the composite function.
'f1.f0' is the location of the resolution function (to be convolved with model).
'f1.f1' is the location of the 'model' function (to be convolved with resolution).
'f1.f1.fn' is the location of the nth product function in the model.

The 'tempPrefix' variable here actually specifies the position of a function within a product function (product is with a potentially supplied temperature correction), if temperature isn't provided, it will be 'f0', else it will be 'f1'.

It would be best in the future if this weren't hard-coded in this manner, I initially considered this to be outside the scope of this PR, since this is always true for now.